build.rs

```diff
@@ -185,6 +186,41 @@ fn main() -> anyhow::Result<()> {
     .join("\n");
     fs::write(&dest_path, conf_decl).unwrap();
 
+    let metadata = MetadataCommand::new()
+        .manifest_path("Cargo.toml")
+        .features(CargoOpt::AllFeatures)
+        .exec()
+        .unwrap();
+
+    let root_package_id = metadata
+        .root_package()
+        .ok_or(anyhow::anyhow!("Failed to get ID of root package"))?
+        .id
+        .clone();
+
+    let viam_modules : Vec<_> = metadata
+        // Obtain the dependency graph from the metadata and iterate its nodes
+        .resolve.as_ref()
+        .ok_or(anyhow::anyhow!("Dependencies were not resolved"))?
+        .nodes.iter()
+        // Until we find the root node..
+        .find(|node| node.id == root_package_id)
+        .ok_or(anyhow::anyhow!("Root package not found in dependencies"))?
+        // Then iterate the root node's dependencies, selecting only those
+        // that are normal dependencies.
+        .deps.iter()
+        .filter(|dep| dep.dep_kinds.iter().any(|dk| dk.kind == cargo_metadata::DependencyKind::Normal))
+        // And which have a populated `package.metadata.com.viam` section in their Cargo.toml
+        // which has `module = true`
+        .filter(|dep| metadata[&dep.pkg].metadata["com"]["viam"]["module"].as_bool().unwrap_or(false))
+        .collect();
+
+    let mut modules_rs_content = String::new();
+    let module_name_seq = viam_modules.iter().map(|m| m.name.replace("-", "_")).collect::<Vec<_>>().join(", \n\t");
+    modules_rs_content.push_str(&format!("generate_register_modules!(\n\t{}\n);\n", module_name_seq));
```

Comment:
@npmenard - If there is a templating something you would prefer I use for this rather than hand generating `\n\t` and such, please just let me know, but it didn't feel warranted for something this simple.